rollup-node.md

```diff
@@ -0,0 +1,285 @@
+
+```
+PayloadAttributesOPV1: {
+	timestamp: QUANTITY
+	random: DATA (32 bytes)
+	suggestedFeeRecipient: DATA (20 bytes)
+	transactions: array of DATA
+}
+```
+
+The type notation used here refers to the [HEX value encoding] used by the
+[Ethereum JSON-RPC API specification][JSON-RPC-API], as this structure will need to
+be sent over JSON-RPC. `array` refers to a JSON array.
+
+[HEX value encoding]: https://eth.wiki/json-rpc/API#hex-value-encoding
+[JSON-RPC-API]: https://eth.wiki/json-rpc/API
+
+The object properties must be set as follows:
+
+- `timestamp` is set to the current [unix time] (number of elapsed seconds since
+  00:00:00 UTC on 1 January 1970), rounded to the closest multiple of 2 seconds.
+- `random` is currently set to all zero bytes. After [the merge][merge], the randomness
+  value exposed for the L1 block will be copied in this property.
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, RLP-encoded in the [EIP-2718]
+  format (i.e. as a sequence of a single byte and a byte array), of the allowed
+  types. We specify how to select the transactions below.
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+> For details on RLP encoding, refer to the yellow paper, or look at [this
+> hyperlinked implementation][encode-tx].
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
+
+In particular, we allow the following two transaction types:
+- [EIP-1559]: the current "regular" Ethereum L1 transaction type, which
+  separates gas fees between a base fee and a tip
+- the *[L1 attributes transaction]*, see below
+
+[EIP-1559]: https://eips.ethereum.org/EIPS/eip-1559
+
+Historical Optimistic Ethereum transactions also include [EIP-155] transactions,
+which are not accepted by the network anymore. This is not an [EIP-2718] typed
+transaction, but can be distinguished from those by the fact that their
+RLP-encoding's first byte will always be `>= 0xC0`.
+
+[EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+
+#### Payload Transaction Format
+
+The `transactions` array is filled with the deposits, prefixed by the (single)
+[L1 attributes transaction]. The deposits (which must be similarly RLP-encoded)
+are simply copied byte-for-byte — it is the role of the [execution engine] to
+reject invalidly-formatted transactions.
+
+> **TODO** must offer some precisions on the format of deposits: sender,
+> receivers both in-tx-as-encoded, and on-L2-tx. What about the fees?
+
+The Optimistic Ethereum specific *[L1 attributes transaction]* has the following
+[EIP-2718]-compatible format: `0x7E || [block_number, timestamp, basefee]`
```

Comment:
Wasn't @maurelian going to specify the deposits? We should separate this out, and keep it in the deposits specification doc.

Reply:
Yeah, this could be replaced by a link to that spec. Seemed like a simple enough strawman to put in.

Reply:
Actually revisiting this, this is wrong, this transaction does not appear at all on the L1 side since it is derived from L1 block attributes. Should probably be specified in the execution engine instead. Needs to be mentionned here too in any case, so duplication is warranted. We can change it later, obviously.